---
title: "PLPA 6820 Final Project"
author: "SA Donohoo"
date: "2025-04-24"
output:
  word_document:
    toc: true
  html_document:
    toc_float: true
  md_document:
    variant: gfm
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE,warning = FALSE, message = FALSE)
```
### 01 Linux Scripts
This folder contains bash scripts and programs used in Linux for data processing. 

Raw reads were demultiplexed and filtered using *process_radtags* in Stacks 2.66 (Rochette et al., 2019) and the 01_Clone_Filtering.sh script. Demultiplexed reads were processed and assembled using the *denovo_map.pl* pipeline in Stacks. Parameters were determined following Paris et al. (2017). Assembled 3RAD loci were filtered using *populations* within Stacks.

A complete dataset, including all three species from the Duck River, using multiple single-nucleotide polymorphisms (SNPs) per locus was created for analyses that do not require unlinked loci. A complete dataset with only a single SNP per locus retained was used for analyses that assume loci are unlinked. Multiple and single SNP per locus datasets were also generated for each distinct, lineage.

```{r 01_Linux, echo=TRUE}
# Check for Scripts in Folder
list.files("01_Linux_Scripts/")
```


### 02 Genetic Diversity Estimates
For each lineage, observed heterozygosity (Ho), expected heterozygosity (He), nucleotide diversity (π), and inbreeding coefficients (FIS) were calculated by *populations* using the lineage-specific multiple SNPs per locus datasets. 

```{r 02_Genetic_Estimates, echo=TRUE}
# Load Data from the Multiple SNPS Sumstats_Summary File produced by Stacks
Dutt.sumstats <- read.table("02_Genetic_Diversity_Estimates/Duttoniana_min_max_1pop_Multi.sumstats_summary.tsv", header = TRUE, na.strings = "NA")
Ful.sumstats <- read.table("02_Genetic_Diversity_Estimates/Fuliginosa_min_max_DAPC_4pops_Multi.sumstats_summary.tsv", header = TRUE, na.strings = "NA")
Gent.sumstats <- read.table("02_Genetic_Diversity_Estimates/Geniculata_min_max_NBA_2pop_Multi.sumstats_summary.tsv", header = TRUE, na.strings = "NA")
```

#### 02 Genetic Diversity Estimates - *Lithasia duttoniana*
```{r r 02_Genetic_Estimates Lithasia duttoniana, echo=TRUE}
# Split by Population
Dutt.sumstats$Pop_ID<-factor(Dutt.sumstats$Pop_ID, levels=unique(Dutt.sumstats$Pop_ID))
Dutt.sumstats_list<-split(Dutt.sumstats,Dutt.sumstats$Pop_ID)

#Calculate Averages for each Population
Dutt.stats_table=data.frame()
for (i in Dutt.sumstats_list){
  pop_data<-as.data.frame(i)
  pop<-pop_data$Pop_ID
  Ho<-round(mean(pop_data[,9]),digits=4)
  Ho_SE<-round((pop_data[,11]),digits=5)
  He<-round(mean(pop_data[,15]),digits=4)
  He_SE<-round((pop_data[,16]),digits=5)
  Pi<-round(mean(pop_data[,21]),digits=4)
  Pi_SE<-round((pop_data[,22]),digits=5)
  Fis<-round(mean(pop_data[,24]),digits=4)
  Fis_SE<-round((pop_data[,25]),digits=5)
  Dutt.stats_table<-rbind(Dutt.stats_table,list(pop,Ho,Ho_SE,He,He_SE,Pi,Pi_SE,Fis,Fis_SE))
}

# Rename Columns
names(Dutt.stats_table)<-c("Population","Ho","Ho_SE","He","He_SE","Pi","Pi_SE","Fis","Fis_SE")

# View Table
Dutt.stats_table
```

#### 02 Genetic Diversity Estimates - *Lithasia fuliginosa*
```{r 02_Genetic_Estimates Lithasia fuliginosa, echo=TRUE}
# Split by Population
Ful.sumstats$Pop_ID<-factor(Ful.sumstats$Pop_ID, levels=unique(Ful.sumstats$Pop_ID))
Ful.sumstats_list<-split(Ful.sumstats,Ful.sumstats$Pop_ID)

#Calculate Averages for each Population
Ful.stats_table=data.frame()
for (i in Ful.sumstats_list){
  pop_data<-as.data.frame(i)
  pop<-pop_data$Pop_ID
  Ho<-round(mean(pop_data[,9]),digits=4)
  Ho_SE<-round((pop_data[,11]),digits=5)
  He<-round(mean(pop_data[,15]),digits=4)
  He_SE<-round((pop_data[,16]),digits=5)
  Pi<-round(mean(pop_data[,21]),digits=4)
  Pi_SE<-round((pop_data[,22]),digits=5)
  Fis<-round(mean(pop_data[,24]),digits=4)
  Fis_SE<-round((pop_data[,25]),digits=5)
  Ful.stats_table<-rbind(Ful.stats_table,list(pop,Ho,Ho_SE,He,He_SE,Pi,Pi_SE,Fis,Fis_SE))
}

# Rename Columns
names(Ful.stats_table)<-c("Population","Ho","Ho_SE","He","He_SE","Pi","Pi_SE","Fis","Fis_SE")

# View Table
Ful.stats_table
```

#### 02 Genetic Diversity Estimates - *Lithasia geniculata*
```{r 02_Genetic_Estimates Lithasia geniculata, echo=TRUE}
# Split by Population
Gent.sumstats$Pop_ID<-factor(Gent.sumstats$Pop_ID, levels=unique(Gent.sumstats$Pop_ID))
Gent.sumstats_list<-split(Gent.sumstats,Gent.sumstats$Pop_ID)

#Calculate Averages for each Population
Gent.stats_table=data.frame()
for (i in Gent.sumstats_list){
  pop_data<-as.data.frame(i)
  pop<-pop_data$Pop_ID
  Ho<-round(mean(pop_data[,9]),digits=4)
  Ho_SE<-round((pop_data[,11]),digits=5)
  He<-round(mean(pop_data[,15]),digits=4)
  He_SE<-round((pop_data[,16]),digits=5)
  Pi<-round(mean(pop_data[,21]),digits=4)
  Pi_SE<-round((pop_data[,22]),digits=5)
  Fis<-round(mean(pop_data[,24]),digits=4)
  Fis_SE<-round((pop_data[,25]),digits=5)
  Gent.stats_table<-rbind(Gent.stats_table,list(pop,Ho,Ho_SE,He,He_SE,Pi,Pi_SE,Fis,Fis_SE))
}

# Rename Columns
names(Gent.stats_table)<-c("Population","Ho","Ho_SE","He","He_SE","Pi","Pi_SE","Fis","Fis_SE")

# View Table
Gent.stats_table
```

### 03 Population Structure Analyses - AMOVA
An analysis of molecular variance (AMOVA; Excoffier et al., 1992) was conducted using the multiple SNPs per locus dataset for *Lithasia fuliginosa* and *Lithasia geniculata* to test for population structure among collection sites. Using the poppr.amova function in the R package ADEGENET (Jombart, 2008; Kamvar et al., 2014), individuals were stratified by collection site.

```{r 03_AMOVA_Analysis, echo=TRUE}
# Load packages
library(poppr)

# Load Data from Multiple HAPS Genepop File produced by Stacks
Ful.geneid = read.genepop("03_Population_Structure_AMOVA/Fuliginosa_min_max_DAPC_4pops_Multi.haps.gen")
Gent.geneid = read.genepop("03_Population_Structure_AMOVA/Geniculata_min_max_NBA_2pop_Multi.haps.gen")

# Create a function that conducts an AMOVA using Poppr
Amova.Poppr <- function(geneid){

# Create Strata from Population Information in Geneid
genclone<-as.genclone(geneid)
genclone$strata
strata(genclone)<-(as.data.frame(geneid$pop))

# Run an AMOVA
genclone
genclone$strata
amova.pop = poppr.amova(genclone, ~geneid.pop, cutoff = 0.5, method = "ade4")
}
```

#### 03 Population Structure Analysis - AMOVA - *Lithasia fuliginosa*
```{r 03_AMOVA_Analysis_L_fuliginosa, echo=TRUE}
# Lithasia geniculata AMOVA
Ful.amova <- Amova.Poppr(Ful.geneid)
# View AMOVA Results
Ful.amova
```

#### 03 Population Structure Analysis - AMOVA - *Lithasia geniculata*
```{r 03_AMOVA_Analysis_L_geniculata, echo=TRUE}
# Lithasia geniculata AMOVA
Gent.amova <- Amova.Poppr(Gent.geneid)
# View AMOVA Results
Gent.amova
```

### 04 Population Structure Analyses - DAPC
Discriminant analysis of principal components was conducted in the R package ADEGENET using the multi-species multiple-snp dataset produced by STACKS. The best-fit number of genetic clusters (K) was determined using K-means clustering and Bayesian information criteria (Supporting Information); however, the number of principal component (PC) axes used in the DAPC analysis followed Patterson, Price, and Reich (2006) K - 1 rule.

``` {r 04_DAPC_Analysis, echo=TRUE}
# Load packages
library(adegenet)
library(ape)
library(poppr)
library(na.tools)

# Load Lithasia.data from Multiple SNP Genepop File produced by Stacks
Lithasia.data <- read.genepop(file="04_Population_Structure_DAPC/Lithasia_Duck_min_max_NBA_NoOut_DAPC3.snps.gen")

# Find the Best-Fit # of Principle Components (Genetic Clusters)
grp <- find.clusters(Lithasia.data, max.n.clust=30, n.pca = 500, n.clust = 7)

# Run DAPC Using the Best-Fit # of Clusters, but Apply (NAME) K-1
dapc2 <- dapc(Lithasia.data,grp$grp, n.pca = 6, n.da = 5)

# Create an Updated Populations List from the DAPC Results
write.csv(grp[["grp"]], "04_Population_Structure_DAPC/DuckRiver_DAPC_k7.csv")

# Create a Color-Blind Friendly Palette
cbbPalette <- c("#56B4E9","#E69F00", "#0072B2", "#009E73", "#F0E442", "#D55E00", "#CC79A7", "gray")

# Graph the DAPC Results. Uses Built in Graphing Functions in the DAPC Package.
# Initial Scatterplot with Cluster Assignments
scatter(dapc2, scree.da=FALSE, bg="white", pch=20, cell=0,
        cstar=0, col=cbbPalette, solid=.4, cex=3,clab=0, leg=TRUE, posi.leg = "bottomright")
```

### 05 Population Structure Analyses - LEA Admixture
Genetic admixture among all lineages was assessed with the snmf function in the R package LEA (Frichot & François, 2015) using the single-snp datasets produced by STACKS because the *snmf* method assumes that SNPs are unlinked.

```{r 05_LEA_Admixture_Analysis, echo=TRUE, results = 'hide'}

# Load Packages
library(LEA)
library(BiocManager)
library(ggplot2)
library(tidyverse)

# Load data from Single SNP VCF produced by Stacks
Dutt.input.file <- "05_Population_Structure_Admixture/Duttoniana_min_max_NBA_Single.snps.vcf"
Gent.input.file <- "05_Population_Structure_Admixture/Geniculata_min_max_2Pops_NBA_Single.snps.vcf"
Ful.input.file <- "05_Population_Structure_Admixture/Fuliginosa_min_max_DAPC_4pops_Single.snps.vcf"

# Load in metadata for each Species
Dutt.Metadata <- read.csv("05_Population_Structure_Admixture/Duttoniana_Metadata.csv", header = TRUE, na.strings = "NA")
Gent.Metadata <- read.csv("05_Population_Structure_Admixture/Geniculata_Metadata.csv", header = TRUE, na.strings = "NA")
Ful.Metadata <- read.csv("05_Population_Structure_Admixture/Fuliginosa_Metadata.csv", header = TRUE, na.strings = "NA")

# Create a function that finds the best fit K-value for LEA Admixture Analysis.
Admixture.Lea.K.qmatrix <- function(Input.File, Kvalue){
  Geno <- vcf2geno(Input.File, force = TRUE)
  obj.snmf_100 = snmf(Geno, K = Kvalue, project = "new", alpha = 100, 
                      tolerance = 0.0000000001, repetitions = 10, 
                      entropy=TRUE, ploidy = 2)
  ce <- cross.entropy(obj.snmf_100, K = Kvalue)
  best <- which.min(ce)
  qmatrix <- Q(obj.snmf_100, K = Kvalue, run = best)
}
```

#### *Lithasia duttoniana* Admixture Analysis

```{r 05_LEA_Admixture_Analysis L. duttoniana, echo=TRUE, results = 'hide'}
# Run LEA Admixture for Lithasia duttoniana in the Duck River, TN.
# Best fit K-value is 2.
Dutt.qmatrix.K2 <- Admixture.Lea.K.qmatrix(Dutt.input.file, 2)
```

```{r 05_LEA_Admixture_Analysis L. duttoniana P2, echo=TRUE}
# Convert Vector into DataFrame
Dutt.qmatrix.K2 <- data.frame(Dutt.qmatrix.K2)

# Add Specimen Column from Metadata to Qmatrix for Future Joining
Dutt.qmatrix.K2$Specimen <- Dutt.Metadata$Specimen

# Combine with Metadata DataFrame
Dutt.LEA.K2.Join <- full_join(Dutt.Metadata, Dutt.qmatrix.K2, by = "Specimen")

# Convert from Wide to Long Format
Dutt.LEA.K2.Long <- Dutt.LEA.K2.Join %>% 
  pivot_longer(
    cols = `V1`:`V2`, 
    names_to = "Admixure",
    values_to = "Amount"
  )

# Convert the Column Site into a Factor to Allow for Faceting
Dutt.LEA.K2.Long$Site <- as.factor(Dutt.LEA.K2.Long$Site)

# Using Levels, Reorder Sites so They Run Downstream to UpStream
Dutt.LEA.K2.Long$Site <- factor(Dutt.LEA.K2.Long$Site, levels = c("Dyer",
                                                                      "NV",
                                                                      "CV",
                                                                      "KM",
                                                                      "B",
                                                                      "IP",
                                                                      "HWY412",
                                                                      "R",
                                                                      "IB",
                                                                      "SP",
                                                                      "FW",
                                                                      "SM",
                                                                      "CB",
                                                                      "HM",
                                                                      "HWY31",
                                                                      "WB"
))

# Create a Color-Blind Friendly Palette
cbbPalette <- c("#56B4E9","#E69F00", "#0072B2", "#009E73", "#F0E442", "#D55E00", "#CC79A7")

# Graph the Admixture Results. Use Facet to Split into Sections by Site
Dutt.admixture <- ggplot(Dutt.LEA.K2.Long, aes(x = factor(Specimen), y = Amount, fill = Admixure)) +
  geom_col(width = 1) +
  # Facet over the groups
  facet_grid(~ Site, switch = "x", scales = "free", space = "free") +
  # Use these expand arguments to get rid of padding around the axes
  scale_x_discrete(expand = c(0,0), name = "Lithasia duttoniana Sites") +
  scale_y_continuous(expand = c(0,0), name = "Admixture %") +
  scale_fill_manual(values = (cbbPalette), name = "") +
  # We're hiding most of the x-axis and the spacing between panels
  theme(panel.spacing.x = grid::unit(0.05, "cm"),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
        strip.background = element_blank(),
        panel.border = element_rect(fill = NA, colour = "black"),
        legend.position="")
```

#### Results of LEA Admixture Analysis for *Lithasia duttoniana*

```{r 05_LEA_Admixture_Analysis L. duttoniana P3, echo=FALSE}
# View Admixture Graph
Dutt.admixture
```

#### *Lithasia fuliginosa* Admixture Analysis

```{r 05_LEA_Admixture_Analysis L. fuliginosa, echo=TRUE, results = 'hide'}
# Run LEA Admixture for Lithasia fuliginosa in the Duck River, TN.
# Best fit K-value is 2.
Ful.qmatrix.K4 <- Admixture.Lea.K.qmatrix(Ful.input.file, 4)
```

```{r 05_LEA_Admixture_Analysis L. fuliginosa P2, echo=TRUE}
# Convert Vector into DataFrame
Ful.qmatrix.K4 <- data.frame(Ful.qmatrix.K4)

# Add Specimen Column from Metadata to Qmatrix for Future Joining
Ful.qmatrix.K4$Specimen <- Ful.Metadata$Specimen

# Combine with Metadata DataFrame
Ful.LEA.K4.Join <- full_join(Ful.Metadata, Ful.qmatrix.K4, by = "Specimen")

# Convert from Wide to Long Format
Ful.LEA.K4.Long <- Ful.LEA.K4.Join %>% 
  pivot_longer(
    cols = `V1`:`V4`, 
    names_to = "Admixure",
    values_to = "Amount"
  )

# Convert the Column Site into a Factor to Allow for Faceting
Ful.LEA.K4.Long$Site <- as.factor(Ful.LEA.K4.Long$Site)

# Using Levels, Reorder Sites so They Run Downstream to UpStream
Ful.LEA.K4.Long$Site <- factor(Ful.LEA.K4.Long$Site, levels = c("D",
                                                                    "SP",
                                                                    "FW",
                                                                    "SM",
                                                                    "C",
                                                                    "HDM",
                                                                    "LM",
                                                                    "31",
                                                                    "HM",
                                                                    "WB",
                                                                    "SR",
                                                                    "231",
                                                                    "41A",
                                                                    "TF",
                                                                    "269",
                                                                    "P"
))

# Create a Color-Blind Friendly Palette
cbbPalette <- c("#56B4E9","#E69F00", "#009E73", "#F0E442", "#CC79A7", "#0072B2", "#D55E00")

# Graph the Admixture Results. Use Facet to Split into Sections by Site
Ful.admixture <- ggplot(Ful.LEA.K4.Long, aes(x = factor(Specimen), y = Amount, fill = Admixure)) +
  geom_col(width = 1) +
  # Facet over the groups
  facet_grid(~ Site, switch = "x", scales = "free", space = "free") +
  # Use these expand arguments to get rid of padding around the axes
  scale_x_discrete(expand = c(0,0), name = "Lithasia fuliginosa Sites") +
  scale_y_continuous(expand = c(0,0), name = "Admixture %") +
  scale_fill_manual(values = (cbbPalette), name = "") +
  # We're hiding most of the x-axis and the spacing between panels
  theme(panel.spacing.x = grid::unit(0.05, "cm"),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
        strip.background = element_blank(),
        panel.border = element_rect(fill = NA, colour = "black"),
        legend.position="")
```

#### Results of LEA Admixture Analysis for *Lithasia fuliginosa*

```{r 05_LEA_Admixture_Analysis L. fuliginosa Part 3, echo=FALSE}
# View Admixture Graph
Ful.admixture
```

#### *Lithasia geniculata* Admixture Analysis

```{r 05_LEA_Admixture_Analysis L. geniculata, echo=TRUE, results = 'hide'}
# Run LEA Admixture for Lithasia fuliginosa in the Duck River, TN.
# Best fit K-value is 3.
Gent.qmatrix.K2 <- Admixture.Lea.K.qmatrix(Gent.input.file, 2)
```

```{r 05_LEA_Admixture_Analysis L. geniculata P2, echo=TRUE}
# Convert Vector into DataFrame
Gent.qmatrix.K2 <- data.frame(Gent.qmatrix.K2)

# Add Specimen Column from Metadata to Qmatrix for Future Joining
Gent.qmatrix.K2$Specimen <- Gent.Metadata$Specimen

# Combine with Metadata DataFrame
Gent.LEA.K2.Join <- full_join(Gent.Metadata, Gent.qmatrix.K2, by = "Specimen")

# Convert from Wide to Long Format
Gent.LEA.K2.Long <- Gent.LEA.K2.Join %>% 
  pivot_longer(
    cols = `V1`:`V2`, 
    names_to = "Admixure",
    values_to = "Amount"
  )

# Convert the Column Site into a Factor to Allow for Faceting
Gent.LEA.K2.Long$Site <- as.factor(Gent.LEA.K2.Long$Site)

# Using Levels, Reorder Sites so They Run Downstream to UpStream
Gent.LEA.K2.Long$Site <- factor(Gent.LEA.K2.Long$Site, levels = c("Dyer",
                                                                      "NV",
                                                                      "CV",
                                                                      "TB",
                                                                      "LL",
                                                                      "BL",
                                                                      "KM",
                                                                      "FM",
                                                                      "B",
                                                                      "CS",
                                                                      "IP",
                                                                      "R"
))

# Create a Color-Blind Friendly Palette
cbbPalette <- c("#56B4E9","#E69F00", "#CC79A7", "#0072B2", "#009E73", "#F0E442", "#D55E00")

# Graph the Admixture Results. Use Facet to Split into Sections by Site
Gent.admixture <- ggplot(Gent.LEA.K2.Long, aes(x = factor(Specimen), y = Amount, fill = Admixure)) +
  geom_col(width = 1) +
  # Facet over the groups
  facet_grid(~ Site, switch = "x", scales = "free", space = "free") +
  # Use these expand arguments to get rid of padding around the axes
  scale_x_discrete(expand = c(0,0), name = "Lithasia geniculata Sites") +
  scale_y_continuous(expand = c(0,0), name = "Admixture %") +
  scale_fill_manual(values = (cbbPalette), name = "") +
  # We're hiding most of the x-axis and the spacing between panels
  theme(panel.spacing.x = grid::unit(0.05, "cm"),
        axis.text.x = element_blank(),
        axis.ticks.x = element_blank(),
        strip.background = element_blank(),
        panel.border = element_rect(fill = NA, colour = "black"),
        legend.position="")
```

#### Results of LEA Admixture Analysis for *Lithasia geniculata*

```{r 05_LEA_Admixture_Analysis L. geniculata Part 3, echo=FALSE}
# View Admixture Graph
Gent.admixture
```

### 06 Genetic Co-ancestry - fineRADstructure
Genetic co-ancestry among all lineages was assessed with fineRADstructure in Linux using the multiple-snp datasets produced by STACKS. The results of the fineRADstructure analyses were visualized using scripts modified from the original program. Plots are stored in a seperate folder due to size issues with R consule. fineRADstructure linux commands are included in script: 06_Genetic_Coancestry_Linux_Commands.sh

```{r 06_Genetic_Coancestry_Analysis, echo=TRUE, results = 'hide'}
### 1) EDIT THE PATH TO YOUR COPY of FinestructureLibrary.R
source("06_Genetic_CoAncestry/FinestructureLibrary.R", chdir = TRUE) # read in the R functions, which also calls the needed packages

### 2) EDIT THIS PATH TO WHERE YOU WANT THE PLOTS:
plotsFolder <- "06_Genetic_CoAncestry/RADpainter_R_Plots/"

### 3) EDIT THE FOLLOWING THREE LINES TO PROVIDE PATHS TO THE fineRADstructure OUTPUT 

#### Lithasia duttoniana
Dut.chunkfile<-"06_Genetic_CoAncestry/Duttoniana_min_max_1pop_Multi.haps_chunks.out" ## RADpainter output file
Dut.mcmcfile<-"06_Genetic_CoAncestry/Duttoniana_min_max_1pop_Multi.haps_chunks.mcmc.xml" ## finestructure mcmc file
Dut.treefile<-"06_Genetic_CoAncestry/Duttoniana_min_max_1pop_Multi.haps_chunks.mcmcTree.xml" ## finestructure tree file
Dut.analysisName <- "Lithasia_duttoniana_Genetic_CoAncestry";  maxIndv <- 1000; maxPop<-1000

#### Lithasia fuliginosa
Ful.chunkfile<-"06_Genetic_CoAncestry/Fuliginosa_min_max_DAPC_4pops_Multi.haps_chunks.out" ## RADpainter output file
Ful.mcmcfile<-"06_Genetic_CoAncestry/Fuliginosa_min_max_DAPC_4pops_Multi.haps_chunks.mcmc.xml" ## finestructure mcmc file
Ful.treefile<-"06_Genetic_CoAncestry/Fuliginosa_min_max_DAPC_4pops_Multi.haps_chunks.mcmcTree.xml" ## finestructure tree file
Ful.analysisName <- "Lithasia_fuliginosa_Genetic_CoAncestry";  maxIndv <- 1000; maxPop<-1000

#### Lithasia geniculata
Gent.chunkfile<-"06_Genetic_CoAncestry/Geniculata_min_max_NBA_2pop_Multi.haps_chunks.out" ## RADpainter output file
Gent.mcmcfile<-"06_Genetic_CoAncestry/Geniculata_min_max_NBA_2pop_Multi.haps_chunks.mcmc.xml" ## finestructure mcmc file
Gent.treefile<-"06_Genetic_CoAncestry/Geniculata_min_max_NBA_2pop_Multi.haps_chunks.mcmcTree.xml" ## finestructure tree file
Gent.analysisName <- "Lithasia_geniculata_Genetic_CoAncestry";  maxIndv <- 1000; maxPop<-1000
```

```{r 06_Genetic_Coancestry_Analysis P2, echo=TRUE}
### Create a function to visualize fineRADstructure outputs.
fineRADstructure <- function(chunkfile,mcmcfile,treefile,analysisName){
### 4) EXECUTE THE CODE ABOVE AND THE REST OF THE CODE BELOW
## make some colours
some.colors<-MakeColorYRP() # these are yellow-red-purple
some.colorsEnd<-MakeColorYRP(final=c(0.2,0.2,0.2)) # as above, but with a dark grey final for capped values

###### READ IN THE CHUNKCOUNT FILE
dataraw<-as.matrix(read.table(chunkfile,row.names=1,header=T,skip=1)) # read in the pairwise coincidence 

###### READ IN THE MCMC FILES
mcmcxml<-xmlTreeParse(mcmcfile) ## read into xml format
mcmcdata<-as.data.frame.myres(mcmcxml) ## convert this into a data frame

###### READ IN THE TREE FILES
treexml<-xmlTreeParse(treefile) ## read the tree as xml format
ttree<-extractTree(treexml) ## extract the tree into ape's phylo format

## Reduce the amount of significant digits printed in the posteror assignment probabilities (numbers shown in the tree):
ttree$node.label[ttree$node.label!=""] <-format(as.numeric(ttree$node.label[ttree$node.label!=""]),digits=2)

# convert to dendrogram format
tdend<-myapetodend(ttree,factor=1)

## Now we work on the MAP state
mapstate<-extractValue(treexml,"Pop") # map state as a finestructure clustering
mapstatelist<-popAsList(mapstate) # .. and as a list of individuals in populations
popnames<-lapply(mapstatelist,NameSummary) # population names IN A REVERSIBLE FORMAT (I.E LOSSLESS)
## NOTE: if your population labels don't correspond to the format we used (NAME<number>) YOU MAY HAVE TROUBLE HERE. YOU MAY NEED TO RENAME THEM INTO THIS FORM AND DEFINE YOUR POPULATION NAMES IN popnamesplot BELOW
popnamesplot<-lapply(mapstatelist,NameMoreSummary) # a nicer summary of the populations
names(popnames)<-popnamesplot # for nicety only
names(popnamesplot)<-popnamesplot # for nicety only
popdend<-makemydend(tdend,mapstatelist) # use NameSummary to make popdend
popdend<-fixMidpointMembers(popdend) # needed for obscure dendrogram reasons
popdendclear<-makemydend(tdend,mapstatelist,"NameMoreSummary")# use NameMoreSummary to make popdend
popdendclear<-fixMidpointMembers(popdendclear) # needed for obscure dendrogram reasons


########################
## Plot 1: COANCESTRY MATRIX
fullorder<-labels(tdend) # the order according to the tree
datamatrix<-dataraw[fullorder,fullorder] # reorder the data matrix

tmpmat<-datamatrix 
tmpmat[tmpmat>maxIndv]<-maxIndv #  # cap the heatmap
pdf(file=paste(plotsFolder,analysisName,"-SimpleCoancestry.pdf",sep=""),height=25,width=25)
plotFinestructure(tmpmat,dimnames(tmpmat)[[1]],dend=tdend,cols=some.colorsEnd,cex.axis=1.1,edgePar=list(p.lwd=0,t.srt=90,t.off=-0.1,t.cex=1.2))
dev.off()
}
```

```{r 06_Genetic_CoAncestry P3, echo=TRUE}
# Run fineRADstructure function to visualize outputs
Dutt.fine <- fineRADstructure(Dut.chunkfile,Dut.mcmcfile,Dut.treefile,Dut.analysisName)
Ful.fine <- fineRADstructure(Ful.chunkfile,Ful.mcmcfile,Ful.treefile,Ful.analysisName)
Gent.fine <- fineRADstructure(Gent.chunkfile,Gent.mcmcfile,Gent.treefile,Gent.analysisName)

# Check for PDF files in Folder
list.files("06_Genetic_CoAncestry/RADpainter_R_Plots/")
```

